### PR TITLE
Remove Thread::m_pDomain - always the one app domain

### PR DESCRIPTION
**Pull request by elinor-fung**
We only have one app domain now. The `Thread` constructor was always just initializing its member variable to the current domain.
- Remove `Thread::m_pDomain/GetDomain`
- Remove `AppDomain` parameter on `DebuggerPatchSkip` and checking for mismatch (always the current domain, no mismatch)

### CODE REVIEW

File: src/coreclr/vm/threadsuspend.cpp [hunk 1]
```diff
@@ -4570,7 +4570,7 @@ void Thread::SysResumeFromDebug(AppDomain *pAppDomain)
     while ((thread = ThreadStore::GetThreadList(thread)) != NULL)
     {
         // Only consider resuming threads if they're in the correct appdomain
-        if (pAppDomain != NULL && thread->GetDomain() != pAppDomain)
+        if (pAppDomain != NULL && AppDomain::GetCurrentDomain() != pAppDomain)
```
Comment (lambdageek) [inline src/coreclr/vm/threadsuspend.cpp:4573]: This can't happen?

File: src/coreclr/debug/daccess/dacdbiimpl.cpp [hunk 2]
```diff
@@ -4922,7 +4922,7 @@ VMPTR_AppDomain DacDbiInterfaceImpl::GetCurrentAppDomain(VMPTR_Thread vmThread)
     DD_ENTER_MAY_THROW;
 
     Thread *    pThread    = vmThread.GetDacPtr();
-    AppDomain * pAppDomain = pThread->GetDomain();
+    AppDomain * pAppDomain = AppDomain::GetCurrentDomain();
 
     if (pAppDomain == NULL)
```
Comment (lambdageek) [inline src/coreclr/debug/daccess/dacdbiimpl.cpp:4927]: Can't happen?

**Review by lambdageek — COMMENTED** [review]
seems like `AppDomain::GetCurrentDomain()` can't be null?

File: src/coreclr/debug/ee/debugger.cpp [hunk 3]
```diff
@@ -5828,7 +5828,7 @@ void Debugger::SuspendForGarbageCollectionCompleted()
         InitIPCEvent(ipce1,
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/debug/ee/debugger.cpp:5828]: Thoughts on removing the `AppDomain` argument from `InitIPCEvent()`?

**Review by AaronRobinsonMSFT — APPROVED** [review]
LGTM

**Comment by mikem8361** [comment]
@elinor-fung

>> @mikem8361 do you know if that is (still) the case? I didn't find anything in dotnet/diagnostics that seemed to be directly reading the app domain from that field.
>>
>> The closest thing I found was OFFSETOF__TLS__tls_EETlsData, but I don't see where that is used. https://github.com/dotnet/diagnostics/blob/cbebfd3838eb6adc6f39e47b0786f8f245f34efe/src/shared/inc/tls.h#L19

It is used here:
https://github.com/dotnet/diagnostics/blob/cbebfd3838eb6adc6f39e47b0786f8f245f34efe/src/SOS/Strike/strike.cpp#L4511

It is display the "special thread type" for a managed thread. This part of SOS is really ugly and basically a contract was created between the runtime and SOS.

**Pull request merged**